examples/bank/bank.go

```diff
@@ -57,12 +57,19 @@ type Account struct {
 	Balance int64
 }
 
-func (a Account) encode() ([]byte, error) {
-	return json.Marshal(a)
+func (a Account) encode(retErr *error) []byte {
```

Comment:
this method signature reads much worse to me. Also, in this pattern of error-as-out-parameter wouldn't you end up with only the last error in a chain of operations? That seems Really Bad.